src/Newton.jl

```diff
@@ -109,14 +109,14 @@ function newton(Fhandle, Jhandle, x0, p0, options::NewtonPar; normN = norm, call
 		J = Jhandle(x, p0)
 		d, _, itlinear = options.linsolver(J, f)
 
-		# Update solution: x .= x .- d
-		minus!(x, d)
+		# Update solution:
+		x = x .- d
 
-		copyto!(f, Fhandle(x, p0))
+		f = copy(Fhandle(x, p0))
 		res = normN(f)
 
 		neval += 1
-		push!(resHist, res)
+		history && push!(resHist, res)
```

Comment:
avoiding `push!` by giving the user the option as to whether to store results history; by default `history = false`

Reply:
Hi,

Thank you a lot for the work! It seems to be 2 issues here:
- that the iterable interface does not allow to do bifurcation detection but you seem to have been able to get around it... nicely done.
- the mutating functions which clash with `Zygote`

Reply:
I like the idea for `history`, I would put it as a new boolean field `saveIterations` in `NewtonPar`.

However, the non-mutating version seems disastrous for performances if the arrays are big. 
Could you not do it with a custom vector type and dispatch [those](https://rveltz.github.io/PseudoArcLengthContinuation.jl/dev/#Requested-methods-for-Custom-State-1) ?

Reply:
I've added `saveIterations` to `NewtonPar` in a087b0d

Reply:
regarding your point 2.
since only `copyto!` and `minus!` are affected, can we replace `copyto!(var,new)` with something like `var = copyto!(var,new,inplace=options.inplace)` and pass the boolean `inplace` though `ContinuationPar`? By default it would be `true`

Reply:
regarding point 3. In general you can keep `kwargs...` but the important thing is that you do not pass variables that you want to differentiate as a keyword argument. This is why I had to remove `p = p0` from `newton` at src/Continuation.jl:430